File: services/lucy-dotnet/src/Lucy.AuthPayment.Api/Modules/Payment/Services/IPaymentService.cs
using lucy_auth_payment.Modules.Payment.DTOs;
using lucy_auth_payment.Modules.Payment.Models;

namespace lucy_auth_payment.Modules.Payment.Services;

public interface IPaymentService
{
    Task<Wallet> GetOrCreateWalletAsync(string userId);
    Task<List<WalletTransaction>> GetTransactionsAsync(string userId);
    Task<TopUpOrderResponse> DepositAsync(string userId, DepositRequest request);
    Task<PaymentSetting?> GetMomoSettingAsync();
    Task<PaymentSetting> SaveMomoSettingAsync(SavePaymentSettingRequest request);
    Task<PaymentSetting> UploadMomoQrAsync(IFormFile file);
    Task<List<TopUpOrderResponse>> GetTopUpOrdersAsync(string? status);
    Task<TopUpOrderResponse> ApproveTopUpOrderAsync(string topUpOrderId);
    Task<TopUpOrderResponse> RejectTopUpOrderAsync(string topUpOrderId, string? reason);
    Task<ContentPurchase> PurchaseContentAsync(string buyerUserId, PurchaseContentRequest request);
    Task<LiveAccessTicket> PurchaseLiveAsync(string buyerUserId, PurchaseLiveRequest request);
    Task<Donation> DonateAsync(string fromUserId, DonateRequest request);
    Task<List<WithdrawRequestEntity>> GetWithdrawRequestsAsync(string? status);
    Task<WithdrawRequestEntity> WithdrawAsync(string userId, WithdrawRequest request);
    Task<WithdrawRequestEntity> ApproveWithdrawAsync(string withdrawRequestId);
    Task<WithdrawRequestEntity> RejectWithdrawAsync(string withdrawRequestId, string? rejectReason);
    Task<List<Gift>> GetGiftsAsync();
}

File: lucy-web-stack/lucy-backend/src/main/java/com/lucy/backend/payment/service/IPaymentService.java
package com.lucy.backend.payment.service;

import com.lucy.backend.payment.dto.*;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;

public interface IPaymentService {
    Object getOrCreateWalletAsync(String userId);
    List<Object> getTransactionsAsync(String userId);
    Object depositAsync(String userId, DepositRequest request);
    Object getMomoSettingAsync();
    Object saveMomoSettingAsync(SavePaymentSettingRequest request);
    Object uploadMomoQrAsync(MultipartFile file);
    List<Object> getTopUpOrdersAsync(String status);
    Object approveTopUpOrderAsync(String topUpOrderId);
    Object rejectTopUpOrderAsync(String topUpOrderId, String reason);
    Object purchaseContentAsync(String buyerUserId, PurchaseContentRequest request);
    Object purchaseLiveAsync(String buyerUserId, PurchaseLiveRequest request);
    Object donateAsync(String fromUserId, DonateRequest request);
    List<Object> getWithdrawRequestsAsync(String status);
    Object withdrawAsync(String userId, WithdrawRequest request);
    Object approveWithdrawAsync(String withdrawRequestId);
    Object rejectWithdrawAsync(String withdrawRequestId, String rejectReason);
    List<Object> getGiftsAsync();
}
